### ISSUE TITLE
`kfp-profile-controller` python not found

### ISSUE THREAD

**Issue by gustavosr98** [canonical/kfp-operators#487]
### Bug Description

`kfp-profile-controller` python not found

### To Reproduce

```bash
juju deploy kubeflow --channel 1.8/stable --trust
```

### Environment

Juju 3.1/stable
Microk8s 1.30/stable
Kubeflow 1.8/stable

### Relevant Log Output

```shell
# ----- Juju status
juju status --color| grep -vE "active|started"                                                                                                                                                                                                      g: Sat Jun  1 01:07:22 2024

Model     Controller          Cloud/Region        Version  SLA          Timestamp
kubeflow  microk8s-localhost  microk8s/localhost  3.1.8    unsupported  01:07:23Z

App                        Version                  Status       Scale  Charm                    Channel          Rev  Address         Exposed  Message
katib-db                   8.0.35-0ubuntu0.22.04.1  waiting          1  mysql-k8s                8.0/stable       127  10.152.183.247  no       installing agent
katib-db-manager                                    waiting          1  katib-db-manager         0.16/stable      411  10.152.183.199  no       installing agent
kfp-persistence                                     waiting          1  kfp-persistence          2.0/stable      1291  10.152.183.221  no       installing agent
kfp-profile-controller                              waiting          1  kfp-profile-controller   2.0/stable      1278  10.152.183.22   no       installing agent
kfp-ui                                              maintenance      1  kfp-ui                   2.0/stable      1285  10.152.183.45   no       Reconciling charm: executing component relation:ingress

Unit                          Workload     Agent  Address     Ports          Message
katib-db-manager/0*           waiting      idle   10.1.67.22                 Waiting for relational-db data
katib-db/0*                   maintenance  idle   10.1.67.23                 Initialising mysqld
kfp-persistence/0*            maintenance  idle   10.1.67.30                 Reconciling charm: executing component container:persistenceagent
kfp-profile-controller/0*     waiting      idle   10.1.67.32                 [container:kfp-profile-controller] Waiting for Pebble services (kfp-profile-controller).  If this persists, it could ...

# ----- Logs 
unit-kfp-profile-controller-0: 00:59:17 INFO unit.kfp-profile-controller/0.juju-log Executing component: 'container:kfp-profile-controller'
unit-kfp-profile-controller-0: 00:59:17 ERROR unit.kfp-profile-controller/0.juju-log execute_components caught unhandled exception when executing configure_charm for container:kfp-profile-controller
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-kfp-profile-controller-0/charm/venv/charmed_kubeflow_chisme/components/charm_reconciler.py", line 92, in reconcile
    component_item.component.configure_charm(event)
  File "/var/lib/juju/agents/unit-kfp-profile-controller-0/charm/venv/charmed_kubeflow_chisme/components/component.py", line 50, in configure_charm
    self._configure_unit(event)
  File "/var/lib/juju/agents/unit-kfp-profile-controller-0/charm/venv/charmed_kubeflow_chisme/components/pebble_component.py", line 131, in _configure_unit
    self._update_layer()
  File "/var/lib/juju/agents/unit-kfp-profile-controller-0/charm/venv/charmed_kubeflow_chisme/components/pebble_component.py", line 142, in _update_layer
    container.replan()
  File "/var/lib/juju/agents/unit-kfp-profile-controller-0/charm/venv/ops/model.py", line 1915, in replan
    self._pebble.replan_services()
  File "/var/lib/juju/agents/unit-kfp-profile-controller-0/charm/venv/ops/pebble.py", line 1680, in replan_services
    return self._services_action('replan', [], timeout, delay)
  File "/var/lib/juju/agents/unit-kfp-profile-controller-0/charm/venv/ops/pebble.py", line 1761, in _services_action
    raise ChangeError(change.err, change)
ops.pebble.ChangeError: cannot perform the following tasks:
- Start service "kfp-profile-controller" (cannot start service: exec: "python": executable file not found in $PATH)
----- Logs from task 0 -----
2024-06-01T00:59:17Z ERROR cannot start service: exec: "python": executable file not found in $PATH
-----
unit-kfp-profile-controller-0: 00:59:18 INFO unit.kfp-profile-controller/0.juju-log HTTP Request: GET https://10.152.183.1/apis/apiextensions.k8s.io/v1/customresourcedefinitions "HTTP/1.1 200 OK"
```

**Comment by orfeas-k** [comment]
The cause of this issue was that during https://github.com/canonical/kfp-operators/pull/414 we forgot to update the command to use `python3`, as we did in https://github.com/canonical/kfp-operators/pull/477. The thing that I do not get is why the CI is passing since it's checking the status of the charms, and as this issue mentioned (and I reproduced on juju 3.5), the charm is not going to Active when the workload cannot start. I [reran the kfp bundle tests](https://github.com/canonical/kfp-operators/actions/runs/9223515108/job/25835527500) in a PR:
1. If it fails, it probably means that something changed in the image we use.
2. If it succeeds, I 'll file an issue so we can improve our test coverage.

**Comment by motjuste** [comment]
We at Solutions QA have also seen this bug occur since the latest update ... you'll find the identified occurrences [here](https://solutions.qa.canonical.com/bugs/gh:kfp-profile-controller:487).

**Comment by orfeas-k** [comment]
Thank you @motjuste for adding more information. This issue has now been fixed now (in #501) and is released to `2.0/stable`. New CKF `1.8/stable` should have the fix by default while for existing ones, you can refresh the specific charm to latest changes `2.0/stable`. While doing so, make sure that the resource is also upgraded (juju outputs a relevant message in the refresh command).
